app/assets/javascripts/general.js

```diff
@@ -1,31 +1,46 @@
 $(document).ready(function() {
  // flash message for people coming from other countries
- if(window.location.search.substring(1).search("country_name") == -1) {
-    if (!$.cookie('has_seen_country_message')) {
-  $.ajax({
-    url: "/country_message",
-        dataType: 'html',
-        success: function(country_message){
+ var htmlWrapperFront = [
+                          '<div class="popup popup--popup popup--locality" role="alert" id="locality-popup">',
+                          ' <div class="popup__content">',
+                        ];
+ var htmlWrapperBack = [
+                          '   <a href="#top" class="popup__close js-popup__close" aria-label="close">',
+                          '     <span aria-hidden="true">&times;</span>',
+                          '   </a>',
+                          ' </div>',
+                          '</div>',
+                        ];
+var wholeMessage = '';
+if(window.location.search.substring(1).search("country_name") == -1) {
+  if (!$.cookie('has_seen_country_message')) {
+    $.ajax({
+      url: "/country_message",
+      dataType: 'html',
+      success: function(country_message){
         if (country_message != ''){
-      $('#other-country-notice .popup-content').html(country_message);
-      $('body:not(.front) #other-country-notice').show()
+          wholeMessage = htmlWrapperFront.join('') + country_message + htmlWrapperBack.join('');
+          $('#country-message').html(wholeMessage);
+          $('body:not(.front) #locality-popup').show()
         }
-    }
-      })
-
-     }
- }
-
- // popup messages
- $('#other-country-notice .popup-close').click(function() {
-   $('#other-country-notice').hide('slow');
-   $.cookie('has_seen_country_message', 1, {expires: 365, path: '/'});
-     });
- $('#everypage .popup-close').click(function() {
-   $('#everypage').hide('slow');
-   $.cookie('seen_foi2', 1, { expires: 7, path: '/' });
-   return false;
-   });
+      }
+    })
+  }
+}
+
+ // popups
+$('#locality-popup .js-popup__close').click(function() {
```

Comment:
This doesn't work as the block that creates `$('#locality-popup .js-popup__close')` is async (I think) and may still be running by the time you get here, meaning there's no object to handle the click event for. It fails silently, breaking the close button 😢 

Moving this inside the `success:` block, just after the closing `}` of `if (country_message)` seems to fix it.